File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/fragments/AboutFragment.java
package com.inspiredandroid.linuxcommandbibliotheca.fragments;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.inspiredandroid.linuxcommandbibliotheca.BuildConfig;
import com.inspiredandroid.linuxcommandbibliotheca.R;
import com.inspiredandroid.linuxcommandbibliotheca.misc.Utils;

import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by Simon Schubert on 23.11.15.
 * Talk about the App, Licenses and other projects
 */
public class AboutFragment extends Fragment {

    private FirebaseAnalytics mFirebaseAnalytics;

    public AboutFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_about, container, false);

        ButterKnife.bind(this, view);

        mFirebaseAnalytics = FirebaseAnalytics.getInstance(getContext());
        trackContent();

        return view;
    }


    private void trackContent() {
        if(BuildConfig.DEBUG) {
            return;
        }
        Bundle bundle = new Bundle();
        bundle.putString(FirebaseAnalytics.Param.CONTENT_TYPE, "About");
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.SELECT_CONTENT, bundle);
    }

    @OnClick(R.id.fragment_about_tv_icons8)
    public void startIcons8() {
        try {
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.icons8.com"));
            startActivity(intent);
        } catch (ActivityNotFoundException ignored) {
        }
    }

    @OnClick(R.id.fragment_about_btn_bimo)
    public void startBimo() {
        startAppMarketActivity(Utils.PACKAGE_BIMO);
    }

    @OnClick(R.id.fragment_about_btn_orcgenocide)
    public void startOrcGenocide() {
        startAppMarketActivity(Utils.PACKAGE_ORCGENOCIDE);
    }

    @OnClick(R.id.fragment_about_btn_meowdo)
    public void startMeadow() {
        startAppMarketActivity(Utils.PACKAGE_MEOWDO);
    }

    @OnClick(R.id.fragment_about_btn_remote)
    public void startRemote() {
        startAppMarketActivity(Utils.PACKAGE_LINUXREMOTE);
    }

    @OnClick(R.id.fragment_about_btn_burger)
    public void startBurger() {
        startAppMarketActivity(Utils.PACKAGE_BURGER);
    }

    @OnClick(R.id.fragment_about_btn_rate)
    public void startRateActivity() {
        startAppMarketActivity(Utils.PACKAGE_COMMANDLIBRARY);
    }

    /**
     * Show app in the Play Store. If Play Store is not installed, show it in the browser instead.
     *
     * @param appPackageName package mName
     */
    private void startAppMarketActivity(final String appPackageName) {
        try {
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + appPackageName)));
        } catch (android.content.ActivityNotFoundException e) {
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=" + appPackageName)));
        }
    }
}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/fragments/AboutFragment.kt
package com.inspiredandroid.linuxcommandbibliotheca.fragments

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.inspiredandroid.linuxcommandbibliotheca.R
import com.inspiredandroid.linuxcommandbibliotheca.misc.Constants
import com.inspiredandroid.linuxcommandbibliotheca.misc.showPlayStoreActivity
import kotlinx.android.synthetic.main.fragment_about.*

/* Copyright 2019 Simon Schubert
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
class AboutFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_about, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        btnBraincup.setOnClickListener { context?.showPlayStoreActivity(Constants.PACKAGE_BRAINCUP) }
        btnOrcgenocide.setOnClickListener { context?.showPlayStoreActivity(Constants.PACKAGE_ORCGENOCIDE) }
        btnAncientGenocide.setOnClickListener { context?.showPlayStoreActivity(Constants.PACKAGE_ANCIENTGENOCIDE) }
        btnQuiz.setOnClickListener { context?.showPlayStoreActivity(Constants.PACKAGE_QUIZ) }
        btnRemote.setOnClickListener { context?.showPlayStoreActivity(Constants.PACKAGE_LINUXREMOTE) }
        btnRate.setOnClickListener { context?.showPlayStoreActivity(Constants.PACKAGE_COMMANDLIBRARY) }
        btnIcons8.setOnClickListener { showIcons8Website() }
    }

    private fun showIcons8Website() {
        try {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://www.icons8.com"))
            startActivity(intent)
        } catch (ignored: ActivityNotFoundException) {
        }
    }
}
